app/models/registration_payment.rb

```diff
@@ -9,6 +9,7 @@ class RegistrationPayment < ApplicationRecord
   belongs_to :refunded_registration_payment, class_name: 'RegistrationPayment', optional: true
   has_many :refunding_registration_payments, class_name: 'RegistrationPayment', inverse_of: :refunded_registration_payment, foreign_key: :refunded_registration_payment_id, dependent: :destroy
 
+  after_create :auto_accept_hook, if: -> { registration.auto_accept_in_current_env? && registration.competition.auto_accept_registrations? }
```

Comment:
```suggestion
  delegate auto_accept_in_current_env?, :auto_accept_registrations?, to: :registration
  
  # in registration.rb, you'd also need the following delegate:
  # delegate auto_accept_registrations?, to: :competition

  after_create :auto_accept_hook, if:  [:auto_accept_in_current_env?, :auto_accept_registrations?]
```

Reply:
Committed a version with the comment on line 14/15 (but with the suggested delegation in registration.rb)